### PR TITLE
samples: net: sockets: echo_[server|client]: Add more boards

### PR DESCRIPTION
**Pull request by nandojve**
Add board.conf and board.overlay to run echo_server/client demos using RF2xx transceiver. This helps to validate the IEEE 802.15.4 RF2xx driver and atmel_sam SPI driver. A small topic in the documentation was added to reflect the new configs and explain to user how to build and test.

### CODE REVIEW

File: samples/net/sockets/echo_client/boards/sam_v71_xult.overlay [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+       status = "okay";
+       label = "SPI_RF2XX";
+       cs-gpios = <&portd 25 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+
+       rf2xx@0 {
+               compatible = "atmel,rf2xx";
+               reg = <0x0>;
+               label = "RF2XX_0";
+               spi-max-frequency = <6000000>;
+	       irq-gpios = <&portb 0 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
```
Comment (jukkar) [inline samples/net/sockets/echo_client/boards/sam_v71_xult.overlay]: Indentation

File: samples/net/sockets/echo_server/boards/sam_v71_xult.overlay [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+       status = "okay";
+       label = "SPI_RF2XX";
+       cs-gpios = <&portd 25 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+
+       rf2xx@0 {
+               compatible = "atmel,rf2xx";
+               reg = <0x0>;
+               label = "RF2XX_0";
+               spi-max-frequency = <6000000>;
+	       irq-gpios = <&portb 0 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
```
Comment (jukkar) [inline samples/net/sockets/echo_server/boards/sam_v71_xult.overlay]: ditto

**Review by jukkar — COMMENTED** [review]
Looks good, minor indentation issue in two overlay files.

File: samples/net/sockets/echo_client/boards/sam_v71_xult.overlay (hunk 1 above)
Comment (nandojve) [reply samples/net/sockets/echo_client/boards/sam_v71_xult.overlay]: Fixed on *.overlay files.

**Comment by nandojve** [comment]
I added credits to @benpicco on atsamr21_xpro configs too. He help me with tips to reduce buffers to fit into SAMR21 SoC. Without their tips demo code can't work on that SoC.

**Review by jukkar — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**